### PR TITLE
modification of substations properties, name and country

### PR DESCRIPTION
**Pull request by jamal-khey**
create a pop-up to allow modification of substations properties, name and country

expected: 
- add / update / delete properties via a pop-up
- modify substation name 
- modify substation country

### CODE REVIEW

File: src/components/dialogs/equipements-table/filter-property.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
+import React, { useCallback, useMemo } from 'react';
```
Comment (SlimaneAmar) [inline src/components/dialogs/equipements-table/filter-property.tsx:1]: Add copyright and author

File: src/components/dialogs/equipements-table/filter-property.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+import DeleteIcon from '@mui/icons-material/Delete';
+import IconButton from '@mui/material/IconButton';
```
Comment (SlimaneAmar) [inline src/components/dialogs/equipements-table/filter-property.tsx]: Resolve all warnings

File: src/components/dialogs/equipements-table/site-properties-dialog.tsx [hunk 3]
```diff
@@ -0,0 +1,139 @@
+import React, {
```
Comment (SlimaneAmar) [inline src/components/dialogs/equipements-table/site-properties-dialog.tsx:1]: Same
Add copyright and author and resolve all warnings

File: src/components/spreadsheet/utils/config-tables.js [hunk 4]
```diff
@@ -94,13 +94,24 @@ const generateEditableNumericColumnDefinition = (
         ...extraDef,
     };
 };
+//const onCellClicked = event: CellClickedEvent<TData>
+const onCellSiteCellClicked = (event, callBack) => {
+    switch (event.colDef.id) {
```
Comment (AAJELLAL) [inline src/components/spreadsheet/utils/config-tables.js:99]: Do you really need a switch for just one case?

File: src/services/study/network-modifications.js [hunk 5]
```diff
@@ -1084,6 +1084,57 @@ export function createSubstation(
...
+        const newObj = newPropertiesArray.find(
+            (newObj) => newObj.name === oldObj.name
+        );
+
+        if (!newObj) {
+            propertiesModifications.push({
+                ...oldObj,
+                previousValue: oldObj.value,
+                value: null,
+                // added: false,
```
Comment (AAJELLAL) [inline src/services/study/network-modifications.js:1109]: To remove?

File: src/services/study/network-modifications.js [hunk 6]
```diff
@@ -1084,6 +1084,57 @@ export function createSubstation(
...
+        );
+
+        if (!oldObj) {
+            propertiesModifications.push({
+                ...newObj,
+                added: true,
+                deletionMark: false,
+            });
+        }
+    });
```
Comment (AAJELLAL) [inline src/services/study/network-modifications.js:1134]: Maybe you can refactor this by creating  a single method and use it also  for previousPropertiesArray.

**Review by AAJELLAL — CHANGES_REQUESTED** [review]

File: src/components/spreadsheet/equipment-table.js [hunk 7]
```diff
@@ -29,9 +39,16 @@ export const EquipmentTable = ({
...
     network,
     shouldHidePinnedHeaderRightBorder,
+    onCellClicked,
 }) => {
     const theme = useTheme();
     const intl = useIntl();
+    const [popupSelectEditSiteProperties, setPopupSelectEditSiteProperties] =
+        React.useState(false);
+
+    const [propertiesSite, setPropertiesSite] = React.useState({}); //todo to be renamed with a better name
```
[the remark this answers is not in the document]
Comment (jamal-khey) [reply src/components/spreadsheet/equipment-table.js]: done

File: src/components/dialogs/equipements-table/site-properties-dialog.tsx [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+type IData = {
+    id: number;
+    key: string;
+    value: any;
+};
+
+/**
+ * @author Jamal KHEYYAD <jamal.kheyyad at rte-international.com>
+ */
+const SitePropertiesDialog: React.FC<Props> = ({ data, onDataChanged }) => {
```
[the remark this answers is not in the document]
Comment (jamal-khey) [reply src/components/dialogs/equipements-table/site-properties-dialog.tsx]: done

File: src/components/spreadsheet/equipment-table.js [hunk 9]
```diff
@@ -101,38 +119,116 @@ export const EquipmentTable = ({
...
+            equipmentId,
+            null,
+            false,
+            null,
+            propertiesSiteFormated
+        ).catch((err) => {
+            console.debug(err);
+        });
+
+        //TODO: save data
```
[the remark this answers is not in the document]
Comment (jamal-khey) [reply src/components/spreadsheet/equipment-table.js]: comment removed

[the remark this answers is not in the document]
Comment (jamal-khey) [reply src/components/spreadsheet/equipment-table.js]: comment removed

File: src/components/spreadsheet/equipment-table.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (jamal-khey) [reply src/components/spreadsheet/equipment-table.js]: done

File: src/components/spreadsheet/equipment-table.js [hunk 10]
```diff
@@ -29,9 +39,16 @@ export const EquipmentTable = ({
...
     shouldHidePinnedHeaderRightBorder,
+    onCellClicked,
 }) => {
     const theme = useTheme();
     const intl = useIntl();
+    const [popupSelectEditSiteProperties, setPopupSelectEditSiteProperties] =
+        React.useState(false);
+
+    const [propertiesSite, setPropertiesSite] = React.useState({}); //todo to be renamed with a better name
+    const [siteId, setSiteId] = React.useState(''); //todo to be renamed with a better name
```
[the remark this answers is not in the document]
Comment (jamal-khey) [reply src/components/spreadsheet/equipment-table.js]: done

File: src/components/custom-aggrid/custom-aggrid.tsx [hunk 11]
```diff
@@ -108,6 +108,7 @@ export const CustomAGGrid = React.forwardRef<any, CustomAGGridProps>(
                         loadingOverlayComponentParams
                     }
                     overlayNoRowsTemplate={overlayNoRowsTemplate}
+                    onCellClicked={props.onCellClicked}
```
[the remark this answers is not in the document]
Comment (jamal-khey) [reply src/components/custom-aggrid/custom-aggrid.tsx]: this code has been removed , and a better solution implemented

File: src/components/spreadsheet/equipment-table.js [hunk 12]
```diff
@@ -101,38 +119,116 @@ export const EquipmentTable = ({
...
+
+        //TODO: save data
+        setPopupSelectEditSiteProperties(false);
+        setSiteId('');
+    };
+
+    const handleOnClickOnCell = (params) => {
+        if (onCellClicked != null) {
+            onCellClicked(params, () => {
+                setSiteId(params.data.id);
```
[the remark this answers is not in the document]
Comment (jamal-khey) [reply src/components/spreadsheet/equipment-table.js]: code deleted

File: src/services/study/network-modifications.js (hunk 6 above)
Comment (jamal-khey) [reply src/services/study/network-modifications.js]: i didn't understand what part do you mean? this one
```
propertiesModifications.push({
                ...newObj,
                added: true,
                deletionMark: false,
            });
```

or only this part ? 
```
const updatedProperty = newPropertiesArray.find(
            (updatedObj) => updatedObj.name === previousPropertiePair.name
        );
```

File: src/translations/fr.json [hunk 13]
```diff
@@ -30,6 +30,7 @@
     "modifyAccessRights": "Modifier les droits d'accès",
     "modifyAccessRightsError": "Vous ne pouvez pas modifier les droits d'accès d'une étude qui ne vous appartient pas",
     "edit": "Modifier",
+    "editSiteProperties": "Modifier les proprietées du site",
```
[the remark this answers is not in the document]
Comment (jamal-khey) [reply src/translations/fr.json]: done

File: src/components/dialogs/equipements-table/properties-editor.tsx [hunk 14]
```diff
@@ -0,0 +1,223 @@
...
+}
+
+const PropertiesEditor: React.FC<PropertiesEditorProps> = ({
+    data: initialData,
+    validateData,
+}) => {
+    const [data, setData] = useState<IPropertiesData[]>(
+        initialData.map((item, index) => ({ ...item, id: index }))
+    );
+    // const [data, setData] = useState<IPropertiesData[]>(initialData || [{ id: 0, name: '', value: '' }]);
```
Comment (AAJELLAL) [inline src/components/dialogs/equipements-table/properties-editor.tsx:39]: Remove this.

**Review by AAJELLAL — CHANGES_REQUESTED** [review]

File: src/services/study/network-modifications.js [hunk 15]
```diff
@@ -1095,8 +1095,11 @@ export function formatPropertiesForBackend(
     newPropertiesArray
 ) {
     const propertiesModifications = [];
-
-    previousPropertiesArray.forEach((previousPropertiePair) => {
+    //delete elements that have empty name
+    let filtredPreviousPropertiesArray = previousPropertiesArray.filter(
+        (prop) => prop.name !== ''
+    );
```
[the remark this answers is not in the document]
Comment (jamal-khey) [reply src/services/study/network-modifications.js]: done

**Review by AAJELLAL — CHANGES_REQUESTED** [review]
Here are some bugs found while testing:
**Modifying properties** : 
  - "" is accepted as value.
  - By editing the name, the country and the properties only the properties are saved. The name and the country keep the old values.
  
**Light Mode:**
    - The delete icon is not visible.

**Pull request closed without merging**